stdlib/_typeshed/__init__.pyi

```diff
@@ -82,19 +82,22 @@ class SupportsNext(Protocol[_T_co]):
 class SupportsAnext(Protocol[_T_co]):
     def __anext__(self) -> Awaitable[_T_co]: ...
 
+class SupportsBool(Protocol):
+    def __bool__(self) -> bool: ...
```

Comment:
In case you want to support any eccentric implementations that use a different name for `self`, then you could consider making `self` positional only.